src/main.rs

```diff
@@ -60,14 +69,35 @@ async fn dedup(bot: Throttle<Bot>, msg: Message, db: PgPool) -> HandlerResult {
 
     let seen_before = link.or(forward).or(img);
 
-    if let Some(seen_msg_id) = seen_before {
-        debug!(chat_id, message_id, seen_msg_id, "seen before");
-        let msg_link = Message::url_of(msg.chat.id, msg.chat.username(), MessageId(seen_msg_id));
-        let body = msg_link.map_or_else(|| "看过了".to_string(), |link| format!("看过了: {link}"));
-        bot.send_message(msg.chat.id, body)
-            .reply_to_message_id(msg.id)
-            .await?;
+    let Some((seen_msg_id, seen_times)) = seen_before else {
+        // not seen before, acceptable, early return
+        return Ok(());
+    };
+
+    debug!(chat_id, message_id, seen_msg_id, "seen before");
+    let msg_link = Message::url_of(msg.chat.id, msg.chat.username(), MessageId(seen_msg_id));
+
+    let body = msg_link.map_or_else(
+        || make_reply_msg(seen_times),
+        |link| make_reply_msg_with_link(seen_times, link),
+    );
+    bot.send_message(msg.chat.id, body)
+        .reply_to_message_id(msg.id)
+        .await?;
+    Ok(())
+}
+
+fn make_reply_msg(seen_times: i32) -> String {
+    if seen_times == 1 {
+        "看过了!".to_string()
+    } else if (..=5).contains(&seen_times) {
+        format!("看过 {} 次啦！", seen_times)
+    } else {
+        format!("看过 {} 次啦！不要再发啦！", seen_times)
     }
+}
 
-    Ok(())
+fn make_reply_msg_with_link(seem_times: i32, link: impl ToString) -> String {
+    let msg = make_reply_msg(seem_times);
+    format!("{}:{}", msg.trim_end_matches('!'), link.to_string())
```

Comment:
There are some strings that end with "！" instead of "!" in `make_reply_msg`. What if we remove the trailing exclamation mark in `make_reply_msg` to avoid the issue?